sinonjs/sinon
# Fix for issue #1061 - make unsafe header check in fake server configurable.

#### Fix Issue #1061 - Allow unsafe headers check in fake server using http request to be optional.

#### Background (Problem in detail)

There are times when developer wants to test the app with the fake server without it enforcing any cookie rules with regard to their safety. Currently, this is not possible and in such situations sinon becomes unusable because it blindly checks for unsafe headers and rejects any requests that carry them.

#### Solution

-     Make the unsafe headers optionals/configurable
-     Limit the change to local scope - fake server and fake xmlhttprequest
-     Make the change backward compatible i.e. unless overwritten by user, behavior is the same as before

```
let server = sinon.fakeServer.create()
server.unsafeHeadersEnabled = false
```

#### How to verify - mandatory

1. Check out this branch (see github instructions below)
2.     npm install
3.     run all standard tests - none should fail
4.     create a new basic http request test case and set "cookie" header in the request.
5.     config the fakeServer with unsafeHeaderEnabled = false . test should succeed.
6.     config the fakeServer with unsafeHeaderEnabled = true . test should fail with unsafe header rejection.
